### DISCUSSION TITLE
Using component inside of form

### DISCUSSION THREAD

**Discussion by sachbazjan** [react-hook-form/react-hook-form#2672] [Q&A]
Hey guys.
So my form default values look like this:
```
{
   foo1: undefined,
   foo2: undefined,
   foo3: undefined,
   foo4: undefined,
   foo5: undefined,
}
```

and I have the following form (simplified):

```
<form>
   <input name='foo1' />
   <input name='foo2' />
   <input name='foo3 />
  <ExternalInput />
</form>
```
the external input return:

```
<div>
   <input inputRef={register} name='foo4' />
   <input inputRef={register}  name='foo5' />
</div>
```

Change inside the form results in two seperate objects

```
{
   foo1: 'bar',
   foo2: 'bar',
   foo3: 'bar',
}

{
   foo4: 'bar',
   foo5: 'bar',
}
```

The question is how to achieve having the same structure as the default values

There are more atributes on both the form component and ExternalInput like register, Controler, setValue, ... I will modify the code samples if this info isn't enough.

**Comment by bluebill1049** [comment]
codesandbox?
